Provenance-Emu/Provenance
# Fix game resumption when entering cheat codes

### What does this PR do
This pull request prevents Snes9x games (and possibly PSX games) from playing in the background when entering cheat codes, currently entering the view where the codes are entered results in the game playing in the background.

Feedback is welcome as to how the view checking is handled.

### Where should the reviewer start
Look at 912cb0e47e76ec8d59080ab4a1d4c6fe586391c3, the `viewWillDisappear` method is called when entering the `PVCheatsInfoViewController` view from `PVCheatsViewController` and when returning to the game, it should only be called for the latter.

### How should this be manually tested
Open a SNES game in Snes9x and enter a cheat code, the game should not resume in the background when entering a code.
